src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -85,10 +108,9 @@ public static void RegisterInstance(VisualStudioInstance instance)
         /// </param>
         public static void RegisterMSBuildPath(string msbuildPath)
         {
-            var loadedMSBuildAssemblies = AppDomain.CurrentDomain.GetAssemblies().Where(IsMSBuildAssembly);
-            if (loadedMSBuildAssemblies.Any())
+            if (!CanRegister)
```

Comment:
Interestingly, this fixes a minor bug where you can register multiple time so long as no actual assemblies have been loaded into the app domain yet.